src/main/java/httpserver/HttpRouter.java

```diff
@@ -6,17 +6,26 @@
 
     private final Action methodNotAllowed;
     private final Map<String, Map<String, Action>> routes;
+    private final Action badRequest;
 
-    public HttpRouter(Map<String, Map<String, Action>> routes, Action methodNotAllowed) {
-        this.routes = routes;
+    public HttpRouter(HttpRoutes routes, Action methodNotAllowed, Action badRequest) {
+        this.routes = routes.getRoutes();
```

Comment:
I think there is an another opportunity here. We can certainly return the underlying data structure that `Routes` is using. Alternatively, we could hide those details and provide a convenient API for clients of `Routes` to use to grab the matching route from the `Routes`. 

The question to ask here is should `HttpRouter` need to change if we change the way we store routes in `Routes`?

Reply:
Yes, refactored.